markdown_lab/core/config.py

```
@@ -0,0 +1,200 @@
+
+        if config_path.suffix.lower() == '.json':
+            import json
+            with open(config_path, 'w') as f:
+                json.dump(config_dict, f, indent=2)
+        elif config_path.suffix.lower() in ['.yml', '.yaml']:
+            try:
+                import yaml
+                with open(config_path, 'w') as f:
+                    yaml.dump(config_dict, f, default_flow_style=False)
+            except ImportError:
+                raise ImportError("PyYAML is required to save YAML configuration files")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)